src/main/java/com/dataguise/saas/service/ConnectionService.java

```diff
@@ -0,0 +1,450 @@
+        }
+        List<DatabaseMetadata> response = null;
+        try {
+            response = restDriver.getConnectionMetadata(connectionId, moduleType);
+        } catch (DgDrvException e) {
+            logger.error("Error fetching metadata: " + e);
+            throw new BadGatewayException(e.getMessage(), HttpStatus.BAD_GATEWAY);
+        }
+        logger.info("Metadata fetched successfully");
+        return response;
+    }
+
+    /**
+     * Fetch metadata of new connection by connectionDTO while adding a connection.
+     *
+     * @param dgConnection the existing connection Id
+     * @return List of DatabaseMetadata type.
+     */
+    public List<DatabaseMetadata> fetchNewConnectionMetadata(DgConnection dgConnection)
+        throws BadGatewayException, BadRequestException {
+        List<DatabaseMetadata> response = null;
+        if (dgConnection == null || dgConnection.getConnectionType() == null) {
+            throw new BadRequestException(
+                "Connection type must be specified for getting connection Metdata.",
+                HttpStatus.BAD_REQUEST);
+        }
+        try {
+            dgConnection.setModule(
+                restDriver.getModuleTypeUsingSourceType(dgConnection.getConnectionType()));
+            if (dgConnection.getConnectionId() != null) {
+                logger.info("Connection already exists.");
+                /* check if user changed the password or not.*/
+                DgConnection currDbConnection = getConnections(
+                    Optional.of(dgConnection.getConnectionId()))
+                    .get(0).getConnection();
+                if (currDbConnection.getPassword().equals(dgConnection.getPassword())) {
+                    // call existing connection metadata if the password isn't changed.
+                    // Don't send encrypted password from saas ui to re encrypt.
+                    dgConnection.setPassword("");
+                }
+            }
+            dgConnection.setLocation("Cloud");
+
+            response = restDriver.getConnectionMetadata(dgConnection);
+        } catch (DgDrvException e) {
+            logger.error("Error fetching metadata: " + e);
+            throw new BadGatewayException(e.getMessage(), HttpStatus.BAD_GATEWAY);
+        }
+        logger.info("Metadata fetched successfully");
+        return response;
+    }
+
+    private Boolean createDefaultDetectionSchedule(SchedulerDTO schedule, String moduleType)
+        throws BadRequestException, BadGatewayException, InternalServerException {
+        Boolean status;
+        TimeZone tz = TimeZone.getTimeZone("UTC");
+        // Quoted "Z" to indicate UTC, no timezone offset
+        DateFormat df = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSS'Z'");
+        df.setTimeZone(tz);
+        Calendar c = Calendar.getInstance();
+        c.add(Calendar.MINUTE, delayInMins != null ? delayInMins : 0);
+        logger.info("Default scan type is: " + scanType);
```

Comment:
**Reporter**: PMD    
    **Rule**: GuardLogStatement
    **Severity**: ERROR
    **File**: /work/src/main/java/com/dataguise/saas/service/ConnectionService.java L428
    
    Logger calls should be surrounded by log level guards.&#10;&#10;&#10;Best Practices https://pmd.github.io/pmd/pmd_rules_java_bestpractices.html#guardlogstatement
    
*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a-1475670838>*